src/object/obj_class.c

```diff
@@ -1020,6 +1020,112 @@ daos_oclass_fit_max(daos_oclass_id_t oc_id, int domain_nr, int target_nr,
 	return oc ? 0 : -DER_NONEXIST;
 }
 
+int
+dc_set_oclass(daos_handle_t coh, int domain_nr, int target_nr,
+	      daos_ofeat_t ofeats, daos_oclass_hints_t hints,
+	      daos_oclass_id_t *oc_id_p)
+{
+	uint64_t		rf_factor;
+	daos_oclass_id_t	cid = 0;
+	struct daos_obj_class	*oc;
+	struct daos_oclass_attr	ca;
+	uint16_t		shd, rdd;
+	int			grp_size;
+
+	rf_factor = dc_cont_hdl2redunfac(coh);
+	rdd = hints & DAOS_OCH_RDD_MASK;
+	shd = hints & DAOS_OCH_SHD_MASK;
+
+	/** first set a reasonable default based on RF & RDD hint (if set) */
+	switch (rf_factor) {
+	case DAOS_PROP_CO_REDUN_RF0:
+		if (rdd == DAOS_OCH_RDD_RP)
+			cid = OC_RP_2GX;
+		else if (rdd == DAOS_OCH_RDD_EC)
+			cid = OC_EC_2P1G1;
+		else
+			cid = OC_SX;
+		break;
+	case DAOS_PROP_CO_REDUN_RF1:
+		if (rdd == DAOS_OCH_RDD_RP)
+			cid = OC_RP_2GX;
+		else if (rdd == DAOS_OCH_RDD_EC || ofeats & DAOS_OF_ARRAY ||
+			 ofeats & DAOS_OF_ARRAY_BYTE)
```

Comment:
just confirm that does it mean that dc_array obj always map to EC ?

Reply:
by default, yes if the container has RF > 0